sympy/physics/control/lti.py

```diff
@@ -394,7 +430,7 @@ def __sub__(self, other):
             arg_list = [-i for i in list(other.args)]
             return Parallel(self, *arg_list)
         else:
-            raise ValueError("{} cannot be subtracted from a TransferFunction."
+            raise TypeError("{} cannot be subtracted from a TransferFunction."
```

Comment:
```suggestion
            raise ValueError("{} cannot be subtracted from a TransferFunction."
```
This should remain a `ValueError`, no? `TypeError` is raised when an operation or a function is applied to an object of inappropriate type.

Reply:
I think it should be `TypeError`. Here the exception is raised when the type of object we are subtracting from the `TransferFunction` is incompatible. The placeholder, too (in the print expression), logs the type of the object passed. Say I try to subtract a `Symbol` object from a `TransferFunction`, it will prompt an error -

```python
>>> from sympy.physics.control.lti import *
>>> from sympy.abc import *
>>> TransferFunction(1, s, s)
TransferFunction(1, s, s)
>>> _ - a
Traceback (most recent call last):   
  File "<stdin>", line 1, in <module>
  File "d:\sympy\sympy\physics\control\lti.py", line 436, in __sub__
    raise TypeError("{} cannot be subtracted from a TransferFunction."
TypeError: <class 'sympy.core.symbol.Symbol'> cannot be subtracted from a TransferFunction.
```
It makes sense as the type of object passed is incompatible for tf subtraction. Had this exception been raised when both the params were of the same type, there would have been a possibility for `ValueError`.
I used this as the reference - https://docs.python.org/3/library/exceptions.html#TypeError.

I was anyways, planning to revert to the old error types as suggested by @moorepants since this module has already gone through two release cycles. What is your opinion?

Reply:
The errors shouldn't be changed due to backwards incompatibility. It's more trouble for end users to break their code than worring about if the correct error types were originally selected.

Reply:
> The errors shouldn't be changed due to backwards incompatibility. It's more trouble for end users to break their code than worring about if the correct error types were originally selected.

I will revert the error-type changes.